pkg/controllers/cassandra/service/util/util.go

```diff
@@ -0,0 +1,60 @@
+package util
+
+import (
+	"fmt"
+
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	corelisters "k8s.io/client-go/listers/core/v1"
+
+	v1alpha1 "github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
+	"github.com/jetstack/navigator/pkg/controllers/cassandra/util"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func SetStandardServiceAttributes(
+	cluster *v1alpha1.CassandraCluster,
+	service *apiv1.Service,
+) *apiv1.Service {
+	service.SetNamespace(cluster.Namespace)
+	service.SetLabels(util.ClusterLabels(cluster))
+	service.SetOwnerReferences([]metav1.OwnerReference{
+		util.NewControllerRef(cluster),
+	})
+	service.Spec.Selector = util.NodePoolLabels(cluster, "")
+	return service
+}
+
+type ServiceCreator func(cluster *v1alpha1.CassandraCluster) *apiv1.Service
+type ServiceUpdater func(cluster *v1alpha1.CassandraCluster, service *apiv1.Service) *apiv1.Service
+
+func SyncService(
+	cluster *v1alpha1.CassandraCluster,
+	kubeClient kubernetes.Interface,
+	serviceLister corelisters.ServiceLister,
+	createService ServiceCreator,
+	updateService ServiceUpdater,
+) error {
+	svc := createService(cluster)
```

Comment:
Do we need a creator as well as an updater? In my mind, calling `updateService` with an `&apiv1.Service{}` should have the same effect as calling `createService`?

Reply:
Ah - looks like that's what is always happening, but then we pass a Creator/Updater func to this function. Is there any reason for this? Do we behave differently ever? It'd be ideal to simplify it if we can.

Reply:
I need the create function for generating services for the unit tests. But you're right, it can probably be simplified in a future branch.